backend/services/grading_embedding_service.py

```
@@ -0,0 +1,388 @@
+EMBEDDING_MODEL = "text-embedding-3-small"
+EMBEDDING_DIMENSION = 1536
+
+# Chunking configuration optimized for grading materials
+CHUNK_SIZE = 800  # Smaller chunks for better precision in grading criteria
+CHUNK_OVERLAP = 150  # Overlap to maintain context
+MAX_CHUNKS_FOR_RETRIEVAL = 5  # Top 5 chunks for grading context
+
+class GradingEmbeddingService:
+    """Service for processing grading materials with semantic chunking and embeddings"""
+    
+    def __init__(self):
```

Ruff (0.14.1)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)